synapse/events/third_party_rules.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from twisted.internet import defer
+
+
+class ThirdPartyEventRules(object):
+    def __init__(self, hs):
+        self.third_party_rules = None
+
+        self.hs = hs
+
+        module = None
+        config = None
+        try:
+            module, config = hs.config.third_party_event_rules
+        except Exception:
+            pass
```

Comment:
Please just use an if guard here, i.e. `if hs.config.third_party_event_rules: ...`. The `try .. catch` can easily swallow legitimate errors if you're not careful.